### PR TITLE
Support s3.signer.endpoint for nessie

### PR DESCRIPTION
**Pull request by guitcastro**
This PR honor `s3.signer.endpoint` property, fixing the first point  in #1028.

### CODE REVIEW

File: pyiceberg/io/fsspec.py [hunk 1]
```diff
@@ -86,6 +88,10 @@ def s3v4_rest_signer(properties: Properties, request: AWSRequest, **_: Any) -> A
         raise SignError("Signer set, but token is not available")
 
     signer_url = properties.get(S3_SIGNER_URI, properties["uri"]).rstrip("/")
+    signer_endpoint = properties.get(S3_SIGNER_ENDPOINT, None)
+    if signer_endpoint is None:
+        signer_endpoint = properties.get("endpoint", S3_SIGNER_ENDPOINT_DEFAULT_VALUE)
```
Comment (ndrluis) [inline pyiceberg/io/fsspec.py:93]: I'm not sure if I understand what this "endpoint" property key is. Is there any reason why it shouldn't be something like this: `signer_endpoint = properties.get(S3_SIGNER_ENDPOINT, S3_SIGNER_ENDPOINT_DEFAULT_VALUE)`?

Comment (Fokko) [inline pyiceberg/io/fsspec.py:93]: What @ndrluis is suggesting:
```suggestion
    signer_endpoint = properties.get(S3_SIGNER_ENDPOINT, S3_SIGNER_ENDPOINT_DEFAULT_VALUE)
```

File: pyiceberg/io/fsspec.py [hunk 2]
```diff
@@ -94,7 +100,7 @@ def s3v4_rest_signer(properties: Properties, request: AWSRequest, **_: Any) -> A
         "headers": {key: [val] for key, val in request.headers.items()},
     }
 
-    response = requests.post(f"{signer_url}/v1/aws/s3/sign", headers=signer_headers, json=signer_body)
+    response = requests.post(f"{signer_url}/{signer_endpoint.lstrip()}", headers=signer_headers, json=signer_body)
```
Comment (Fokko) [inline pyiceberg/io/fsspec.py:103]: If we want to clean up the property, why not also prune any trailing whitespace?
```suggestion
    response = requests.post(f"{signer_url}/{signer_endpoint.strip()}", headers=signer_headers, json=signer_body)
```

**Comment by guitcastro** [comment]
@Fokko @ndrluis I've applied the code suggestions. Can you please review again?

File: mkdocs/docs/configuration.md [hunk 3]
```diff
@@ -84,6 +84,7 @@ For the FileIO there are several configuration options available:
...
 | s3.signer            | bearer                   | Configure the signature version of the FileIO.                                                                                                                                                                                                            |
 | s3.signer.uri        | http://my.signer:8080/s3 | Configure the remote signing uri if it differs from the catalog uri. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/v1/aws/s3/sign`.                                                                |
+| s3.signer.endpoint   | v1/main/s3-sign          | Configure the remote signing endpoint. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/<s3.singer.endpoint>. Default to v1/aws/s3/sign`.                                                             |
```
Comment (Fokko) [inline mkdocs/docs/configuration.md:87]: A backtick seems to be missing here:
```suggestion
| s3.signer.endpoint   | v1/main/s3-sign          | Configure the remote signing endpoint. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/<s3.singer.endpoint>`. Default to v1/aws/s3/sign`.                                                             |
```

Comment (guitcastro) [reply mkdocs/docs/configuration.md]: Done!

**Review by ndrluis — APPROVED** [review]

File: mkdocs/docs/configuration.md [hunk 4]
```diff
@@ -84,6 +84,7 @@ For the FileIO there are several configuration options available:
...
 | s3.signer            | bearer                   | Configure the signature version of the FileIO.                                                                                                                                                                                                            |
 | s3.signer.uri        | http://my.signer:8080/s3 | Configure the remote signing uri if it differs from the catalog uri. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/v1/aws/s3/sign`.                                                                |
+| s3.signer.endpoint   | v1/main/s3-sign          | Configure the remote signing endpoint. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/<s3.singer.endpoint>`. Default to v1/aws/s3/sign`.                                                             |
```
Comment (kevinjqliu) [inline mkdocs/docs/configuration.md:87]: what happens if im using pyarrow for fileio?

Comment (kevinjqliu) [reply mkdocs/docs/configuration.md:87]: also s/s3.singer.endpoint/s3.signer.endpoint

File: pyiceberg/io/__init__.py [hunk 5]
```diff
@@ -58,6 +58,8 @@
 S3_PROXY_URI = "s3.proxy-uri"
 S3_CONNECT_TIMEOUT = "s3.connect-timeout"
 S3_SIGNER_URI = "s3.signer.uri"
+S3_SIGNER_ENDPOINT = "s3.signer.endpoint"
+S3_SIGNER_ENDPOINT_DEFAULT_VALUE = "v1/aws/s3/sign"
```
Comment (kevinjqliu) [inline pyiceberg/io/__init__.py:62]: nit: convention is `S3_SIGNER_ENDPOINT_DEFAULT`

**Review by kevinjqliu — COMMENTED** [review]
LGTM, added a few minor comments

File: mkdocs/docs/configuration.md (hunk 4 above)
Comment (guitcastro) [reply mkdocs/docs/configuration.md:87]: > what happens if im using pyarrow for fileio?

The property will be ignored

File: pyiceberg/io/__init__.py (hunk 5 above)
Comment (guitcastro) [reply pyiceberg/io/__init__.py]: done.

File: mkdocs/docs/configuration.md (hunk 4 above)
Comment (guitcastro) [reply mkdocs/docs/configuration.md]: fix it.

File: mkdocs/docs/configuration.md [hunk 6]
```diff
@@ -77,13 +77,14 @@ For the FileIO there are several configuration options available:
...
-| s3.signer.uri        | http://my.signer:8080/s3 | Configure the remote signing uri if it differs from the catalog uri. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/v1/aws/s3/sign`.                                                                |
+| s3.signer.uri        | http://my.signer:8080/s3 | Configure the remote signing uri if it differs from the catalog uri. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/<s3.singer.endpoint>`.                                                          |
+| s3.signer.endpoint   | v1/main/s3-sign          | Configure the remote signing endpoint. Remote signing is only implemented for `FsspecFileIO`. The final request is sent to `<s3.singer.uri>/<s3.singer.endpoint>`. (default : v1/aws/s3/sign).                                                            |
```
Comment (kevinjqliu) [inline mkdocs/docs/configuration.md:87]: looks like there are a few other references to `singer`

Comment (guitcastro) [reply mkdocs/docs/configuration.md]: Done, also renamed some in code

**Review by kevinjqliu — APPROVED** [review]

File: mkdocs/docs/configuration.md (hunk 6 above)
Comment (kevinjqliu) [reply mkdocs/docs/configuration.md]: awesome, thanks!

**Review by Fokko — APPROVED** [review]
@guitcastro Can you run `make lint` to fix the style issues:

```
File : error: indent/outdent mismatch: 7.
Warning: Failed formatting content of a yaml code block (line 278 before formatting). Filename: /home/runner/work/iceberg-python/iceberg-python/mkdocs/docs/configuration.md
Warning: Failed formatting content of a python code block (line 32 before formatting). Filename: /home/runner/work/iceberg-python/iceberg-python/mkdocs/docs/how-to-release.md
Warning: Failed formatting content of a python code block (line 1046 before formatting). Filename: /home/runner/work/iceberg-python/iceberg-python/mkdocs/docs/api.md
Warning: Failed formatting content of a python code block (line 1108 before formatting). Filename: /home/runner/work/iceberg-python/iceberg-python/mkdocs/docs/api.md
```

I think this is good to get in 👍

**Comment by guitcastro** [comment]
@Fokko Done!

**Comment by Fokko** [comment]
Thanks @guitcastro for picking this up, and thanks @ndrluis and @kevinjqliu for the review 🙌

**Pull request merged**